packages/tests/tests/filter/peer_management.spec.ts

```diff
@@ -0,0 +1,75 @@
+import { DefaultPubsubTopic, LightNode } from "@waku/interfaces";
+import {
+  createDecoder,
+  createEncoder,
+  DecodedMessage,
+  utf8ToBytes
+} from "@waku/sdk";
+import { expect } from "chai";
+import { describe } from "mocha";
+
+import {
+  afterEachCustom,
+  beforeEachCustom,
+  ServiceNodesFleet
+} from "../../src/index.js";
+import {
+  runMultipleNodes,
+  teardownNodesWithRedundancy
+} from "../filter/utils.js";
+
+describe("Waku Filter: Peer Management: E2E", function () {
+  this.timeout(15000);
+  let waku: LightNode;
+  let serviceNodes: ServiceNodesFleet;
+
+  beforeEachCustom(this, async () => {
+    [serviceNodes, waku] = await runMultipleNodes(
+      this.ctx,
+      undefined,
+      undefined,
+      5
+    );
+  });
+
+  afterEachCustom(this, async () => {
+    await teardownNodesWithRedundancy(serviceNodes, waku);
+  });
+
+  const pubsubTopic = DefaultPubsubTopic;
+  const contentTopic = "/test";
+
+  const encoder = createEncoder({
+    pubsubTopic,
+    contentTopic
+  });
+
+  const decoder = createDecoder(contentTopic, pubsubTopic);
+
+  it("Number of peers are maintained correctly", async function () {
```

Comment:
Does it add any value comparing to other tests? I see it just testing push and subscribe.

Reply:
oh, right, this one is the important part `expect(successes.length).to.be.equal(waku.filter.numPeersToUse);`

can you add, please, a todo at the beginning of the file to replace it with `unit` test later in future. I think we can be fine with it not being e2e, but I'd like to know what you think

also, do you think we can test somehow rainy day scenarios here? I can't think of any now though